bump.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+# First argument should be kubectl version we're bumping to
+# Second argument should be helm version we're bumping to
+set -euo pipefail
+
+PREV_KUBECTL_VERSION=$(echo "$1 - 1" | bc)
+PREV_HELM_VERSION=$(echo "$2 - 1" | bc)
+
+echo "Before running this script, please ensure you have signed in and have valid AWS credentials, and that your Docker daemon is running."
+read -p "Do you have valid credentials in your account && Docker daemon is running?  (Y/n)" hasValidCreds
+if [[ "$hasValidCreds" =~ ^[nN]$ ]]; then
+  echo "Please obtain valid AWS credentials before running this script. Terminating..."
+  exit 1
+fi
+
+# Check if previous version branch exists
+if ! git rev-parse --verify "kubectl-v$PREV_KUBECTL_VERSION/main" >/dev/null 2>&1; then
+  echo "Error: kubectl-v$PREV_KUBECTL_VERSION does not exist. Cannot produce kubectl-v$1"
+  exit 1
+fi
+
+# Check if kubectl version exists
+echo "Checking if kubectl v$1 exists..."
+if ! curl -L -f -s "https://dl.k8s.io/v1.$1.0/bin/linux/amd64/kubectl" > /dev/null; then
+  echo "Error: kubectl v$1 does not exist at https://dl.k8s.io"
+  exit 1
+fi
+echo "It does!"
+
+# Check if corresponding helm version exists
+echo "Checking if Helm v$2 exists..."
+if ! curl -L -f -s "https://get.helm.sh/helm-v3.$2.0-linux-amd64.tar.gz" > /dev/null; then
+  echo "Error: Helm v$2 does not exist at https://dl.k8s.io"
+  exit 1
+fi
+echo "It does!"
+
+echo "Bumping from Kubectl v$PREV_KUBECTL_VERSION to Kubectl v$1..."
+git clean -fqdx .
+
+# Install dependencies
+yarn install --frozen-lockfile
+
+# Make a new branch and clean it
+git checkout -b kubectl-v${1}/main kubectl-v$PREV_KUBECTL_VERSION/main
+
+# Update README.md
+echo "Updating README.md..."
+sed -i "" "s/KubeCtl v1.$PREV_KUBECTL_VERSION/KubeCtl v1.$1/" README.md                        # Update title banner
+sed -i "" "s/Helm Version: 3.$PREV_HELM_VERSION\.[0-9]/Helm Version: 3.$2.0/" README.md        # Update listed Helm version
+sed -i "" "s/Kubectl Version: 1.$PREV_KUBECTL_VERSION\.[0-9]/Helm Version: 1.$1.0/" README.md  # Update listed KubeCtl version
+sed -i "" "s/KubectlV${PREV_KUBECTL_VERSION}Layer/KubectlV${1}Layer/g" README.md               # Replace construct name
+sed -i "" "s/lambda-layer-kubectl-v$PREV_KUBECTL_VERSION/lambda-layer-kubectl-v$1/" README.md  # Replace import
```

Comment:
these are all hardcoded places where we're expecting changes. if we ever update the readme we'd have to update these hardcoded phrases too. is there a way to just grep for `$PREV_KUBECTL_VERSION` and replace it with `$NEW_KUBECTL_VERSION` or something like that?

Reply:
We probably could, yes. I made sure the strictness only changed the required sections as the `CONTRIBUTING.md` guide mentions, but I feel we could definitely just iterate through files that require changes with the main three changes:
```shell
"s/1.$PREV_KUBECTL_VERSION\.[0-9]/1.$1.0/g" file
"s/3.$PREV_HELM_VERSION\.[0-9]/3.$2.0/g" file
"s/KubectlV${PREV_KUBECTL_VERSION}Layer/KubectlV${1}Layer/g" file
```
The only reason I'd suggest we still at least hardcode the files is in case it matches against sections of the code we don't want it looking at, like dependencies in `package.json` for example.